driver/http/winhttp.c

```diff
@@ -0,0 +1,155 @@
+#include "winhttp.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <windows.h>
+#include <winhttp.h>
+
+#include <euicc/interface.h>
+#include <lpac/utils.h>
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(HINTERNET, WinHttpCloseHandle);
+
+#define utf8_to_wide(input, output)                                            \
+    const int output##n = MultiByteToWideChar(CP_UTF8, 0, input, -1, NULL, 0); \
+    _cleanup_free_ wchar_t *output = calloc(output##n, sizeof(wchar_t));       \
+    MultiByteToWideChar(CP_UTF8, 0, input, -1, output, output##n);
+
+static int http_interface_transmit(struct euicc_ctx *ctx, const char *url, uint32_t *rcode, uint8_t **rx,
+                                   uint32_t *rx_len, const uint8_t *tx, uint32_t tx_len, const char **h) {
+    int fret = 0;
+    _cleanup_(WinHttpCloseHandlep) HINTERNET hSession = NULL;
+    _cleanup_(WinHttpCloseHandlep) HINTERNET hConnect = NULL;
+    _cleanup_(WinHttpCloseHandlep) HINTERNET hRequest = NULL;
+    wchar_t hostname[256];
+    wchar_t pathname[1024];
```

Comment:
256 chars limit makes sense for hostname, but 1024 chars limit for path doesn't make sense, most OS's path limit is 4096B.

Reply:
https://github.com/search?q=repo%3Aestkme-group/lpac%20es9p_trans_json&type=code